File: testflow-master/frontend/src/api/aiModel.ts
/**
 * AI模型管理相关API接口
 */
import api from './index'

export interface AIModelCreate {
  name: string
  provider?: string
  model_id: string
  base_url: string
  api_key?: string
  max_tokens?: number
  temperature?: number
  stream_support?: boolean
  is_active?: boolean
}

export interface AIModelUpdate {
  name?: string
  provider?: string
  model_id?: string
  base_url?: string
  api_key?: string
  max_tokens?: number
  temperature?: number
  stream_support?: boolean
  is_active?: boolean
}

export interface AIModelTest {
  message?: string
}

export interface AgentCreate {
  name: string
  type: string
  ai_model_id: number
  prompt_template?: string
  system_prompt?: string
  temperature?: number
  max_tokens?: number
}

export interface AgentUpdate {
  name?: string
  ai_model_id?: number
  prompt_template?: string
  system_prompt?: string
  temperature?: number
  max_tokens?: number
  is_active?: boolean
}

export interface AIModelResponse {
  id: number
  name: string
  provider: string
  model_id: string
  base_url: string
  max_tokens: number
  temperature: number
  stream_support: boolean
  is_active: boolean
  created_at: string
  updated_at: string
}

export interface AIModelTestResponse {
  success: boolean
  message: string
  response?: any
  error?: string
}

export interface AgentResponse {
  id: number
  name: string
  type: string
  type_display: string
  ai_model_name: string
  temperature: number
  max_tokens: number
  is_active: boolean
  created_at: string
  updated_at: string
}

export const aiModelApi = {
  // AI模型管理
  /**
   * 创建AI模型
   */
  createAIModel(data: AIModelCreate): Promise<AIModelResponse> {
    return api.post('/api/ai/models', data)
  },

  /**
   * 获取AI模型详情（包含API密钥）
   */
  getAIModelDetail(id: number): Promise<AIModelResponse & { api_key: string }> {
    return api.get(`/api/ai/models/${id}`)
  },

  /**
   * 获取AI模型列表
   */
  getAIModels(params?: { skip?: number; limit?: number; is_active?: boolean }): Promise<AIModelResponse[]> {
    return api.get('/api/ai/models', { params })
  },

  /**
   * 更新AI模型
   */
  updateAIModel(id: number, data: AIModelUpdate): Promise<AIModelResponse> {
    return api.put(`/api/ai/models/${id}`, data)
  },

  /**
   * 删除AI模型
   */
  deleteAIModel(id: number): Promise<{ message: string }> {
    return api.delete(`/api/ai/models/${id}`)
  },

  /**
   * 测试AI模型
   */
  testAIModel(id: number, data: AIModelTest = {}): Promise<AIModelTestResponse> {
    return api.post(`/api/ai/models/${id}/test`, data)
  },

  // 智能体管理
  /**
   * 创建智能体
   */
  createAgent(data: AgentCreate): Promise<AgentResponse> {
    return api.post('/api/ai/agents', data)
  },

  /**
   * 获取智能体列表
   */
  getAgents(params?: { skip?: number; limit?: number }): Promise<AgentResponse[]> {
    return api.get('/api/ai/agents', { params })
  },

  /**
   * 更新智能体
   */
  updateAgent(agentId: number, data: AgentUpdate): Promise<AgentResponse> {
    return api.put(`/api/ai/agents/${agentId}`, data)
  },

  /**
   * 删除智能体
   */
  deleteAgent(agentId: number): Promise<{ message: string }> {
    return api.delete(`/api/ai/agents/${agentId}`)
  }
}

export default aiModelApi

File: testflow-master/backend/app/schemas/ai_model.py
from pydantic import BaseModel, Field, HttpUrl
from datetime import datetime
from typing import Optional


class AIModelBase(BaseModel):
    """AI模型基础模型"""
    model_config = {"protected_namespaces": ()}
    
    name: str = Field(..., min_length=1, max_length=100, description="模型显示名称")
    model_id: str = Field(..., min_length=1, max_length=100, description="模型ID")
    base_url: str = Field(..., description="API基础地址")
    api_key: str = Field(..., min_length=10, description="API密钥")
    max_tokens: int = Field(default=4000, ge=100, le=128000, description="最大令牌数")
    temperature: float = Field(default=0.7, ge=0.0, le=2.0, description="温度参数")
    is_active: bool = Field(default=True, description="是否激活")


class AIModelCreate(AIModelBase):
    """创建AI模型"""
    pass


class AIModelUpdate(BaseModel):
    """更新AI模型"""
    model_config = {"protected_namespaces": ()}
    
    name: Optional[str] = Field(None, min_length=1, max_length=100, description="模型显示名称")
    model_id: Optional[str] = Field(None, min_length=1, max_length=100, description="模型ID")
    base_url: Optional[str] = Field(None, description="API基础地址")
    api_key: Optional[str] = Field(None, min_length=10, description="API密钥")
    max_tokens: Optional[int] = Field(None, ge=100, le=128000, description="最大令牌数")
    temperature: Optional[float] = Field(None, ge=0.0, le=2.0, description="温度参数")
    is_active: Optional[bool] = Field(None, description="是否激活")


class AIModelResponse(AIModelBase):
    """AI模型响应"""
    model_config = {"protected_namespaces": (), "from_attributes": True}
    
    id: int
    created_at: datetime
    updated_at: datetime


class AIModelTest(BaseModel):
    """AI模型测试"""
    message: str = Field(default="Hello, this is a test message.", description="测试消息")


class AIModelTestResponse(BaseModel):
    """AI模型测试响应"""
    success: bool
    message: str
    response: Optional[dict] = None
    error: Optional[str] = None
